File: 0057-insert-interval/0057-insert-interval.cpp
class Solution {
public:
    vector<vector<int>> insert(vector<vector<int>>& intervals, vector<int>& newInterval) {
        int n = intervals.size();
        int i = 0;
        vector<vector<int>> newIntervals;
        while (i < n && intervals[i][1] < newInterval[0]){
            newIntervals.push_back(intervals[i]);
            i++;
        }
        while (i < n && intervals[i][0] <= newInterval[1]){
            newInterval[0] = min(intervals[i][0],newInterval[0]);
            newInterval[1] = max(intervals[i][1],newInterval[1]);
            i++;
        }
        newIntervals.push_back(newInterval);
        while (i < n){
            newIntervals.push_back(intervals[i]);
            i++;
        }
        return newIntervals;
    }
};

File: 0057-insert-interval/0057-insert-interval.py
class Solution:
    def insert(self, intervals: List[List[int]], newInterval: List[int]) -> List[List[int]]:
        n = len(intervals)
        newIntervals = []   
        i = 0
        while i < n and intervals[i][1] < newInterval[0]:
            newIntervals.append(intervals[i])
            i += 1
        while i < n and intervals[i][0] <= newInterval[1]:
            newInterval[0] = min(newInterval[0],intervals[i][0])
            newInterval[1] = max(newInterval[1],intervals[i][1])
            i += 1
        newIntervals.append(newInterval)
        while i < n:
            newIntervals.append(intervals[i])
            i += 1
        return newIntervals
